src/Tutor.Core/DomainModel/KnowledgeComponents/KCService.cs

```diff
@@ -50,7 +50,9 @@ public Result<List<InstructionalEvent>> GetInstructionalEventsByKnowledgeCompone
         public Result<AssessmentEvent> SelectSuitableAssessmentEvent(int knowledgeComponentId, int learnerId)
         {
             var knowledgeComponentMastery = _ikcRepository.GetKnowledgeComponentMastery(learnerId, knowledgeComponentId);
-            return knowledgeComponentMastery.SelectSuitableAssessmentEvent(_assessmentEventSelector);
+            var result = knowledgeComponentMastery.SelectSuitableAssessmentEvent(_assessmentEventSelector);
+            _ikcRepository.UpdateKCMastery(knowledgeComponentMastery);
```

Comment:
_ikcRepository should be _kcRepository.

As a side note, I see an issue with the design in that we now have to understand which KCM method generates events and requires a call to the repository. The side effect is really not obvious here, leading me to question why the repository is called...

We should probably research if there is a better way to do this. It's not important presently but long term it can easily lead to issues when we forget to save something like this.

As another side note, we should have tests that check if the appropriate events are stored.

Reply:
I definitely agree. I feel like it'll generally be true for KcMastery that calling any of its methods will cause changes, but usually it's a lot more apparent than in this case.
 One quick way to fix this would be to maybe actually separate this into two methods - one stays SelectSuitableAssessmentEvent and finds and returns the AE, and the other is something like RecordAssessmentEventSelection. That makes it obvious that KcMastery has been changed and needs to be saved. Although I do feel like that's kind of odd in a different way, it seems strange to call an object to do something, and then have to notify it that the something happened.